central/node/globaldatastore/datastore_sac_test.go

```diff
@@ -0,0 +1,471 @@
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := c.Context
+			datastore, err := s.globalDatastore.GetClusterNodeStore(ctx, clusterID, false)
+			if c.ExpectedFound {
+				s.NoError(err)
+				s.Require().NotNil(datastore)
+			} else {
+				s.ErrorIs(err, sac.ErrResourceAccessDenied)
+				s.Nil(datastore)
+			}
+		})
+	}
+}
+
+func (s *nodeDatastoreSACSuite) TestGetClusterNodeStoreWriteAccess() {
+	clusterID := testconsts.Cluster2
+
+	cases := testutils.GenericClusterSACWriteTestCases(context.Background(), s.T(), "write", clusterID, "not-"+clusterID, resources.Node)
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := c.Context
+			datastore, err := s.globalDatastore.GetClusterNodeStore(ctx, clusterID, true)
+			if c.ExpectError {
+				s.ErrorIs(err, sac.ErrResourceAccessDenied)
+				s.Nil(datastore)
+			} else {
+				s.NoError(err)
+				s.Require().NotNil(datastore)
+			}
+		})
+	}
+}
+
+func (s *nodeDatastoreSACSuite) TestRemoveClusterNodeStores() {
+	cases := testutils.GenericGlobalSACDeleteTestCases(s.T())
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := s.testContexts[c.ScopeKey]
+
+			err := s.globalDatastore.RemoveClusterNodeStores(ctx, s.testNodeIDs[testconsts.Cluster1][1])
+			s.waitForIndexing()
+
+			if c.ExpectError {
+				s.ErrorIs(err, c.ExpectedError)
+			} else {
+				s.NoError(err)
+			}
+
+			err = s.globalDatastore.RemoveClusterNodeStores(ctx, s.testNodeIDs[testconsts.Cluster2][0], s.testNodeIDs[testconsts.Cluster2][1])
+			s.waitForIndexing()
+
+			if c.ExpectError {
+				s.ErrorIs(err, c.ExpectedError)
+			} else {
+				s.NoError(err)
+			}
+		})
+	}
+}
+
+func (s *nodeDatastoreSACSuite) TestSearchResults() {
+	clusterID := testconsts.Cluster2
+
+	cases := getReadSACMultiNodeTestCases(context.Background(), s.T(), clusterID, "not-"+clusterID, resources.Node)
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := c.Context
+			results, err := s.globalDatastore.SearchResults(ctx, nil)
+			s.NoError(err)
+
+			fetchedNodeIDs := make([]string, 0, len(results))
+			for _, result := range results {
+				fetchedNodeIDs = append(fetchedNodeIDs, result.Id)
+			}
+
+			var expectedNodeIds []string
+			for _, expectedClusterID := range c.ExpectedClusterIds {
+				expectedNodeIds = append(expectedNodeIds, s.testNodeIDs[expectedClusterID]...)
+			}
+
+			s.ElementsMatch(expectedNodeIds, fetchedNodeIDs)
+		})
+	}
+}
+
+func (s *nodeDatastoreSACSuite) TestSearchRawNodes() {
+	clusterID := testconsts.Cluster2
+
+	cases := getReadSACMultiNodeTestCases(context.Background(), s.T(), clusterID, "not-"+clusterID, resources.Node)
+	for name, c := range cases {
+		s.Run(name, func() {
+			ctx := c.Context
+			rawNodes, err := s.globalDatastore.SearchRawNodes(ctx, nil)
+			s.NoError(err)
+
+			fetchedNodeIDs := make([]string, 0, len(rawNodes))
+			for _, rawNode := range rawNodes {
+				fetchedNodeIDs = append(fetchedNodeIDs, rawNode.Id)
```

Comment:
I think it would make sense to test that the ClusterId field from the retrieved node does match that of the injected node.

Reply:
I have made similar changes here as in the datastore. I'm checking that ClusterId is as expected to be for that node.